tests/integration/suite/test_users.py

```diff
@@ -74,3 +74,53 @@ def check_annotation():
         "clusterscoped.controller.cattle.io/grb-sync_fake"
         not in grb1["metadata"]["finalizers"]
     )
+
+
+def test_roletemplate_finalizer_cleanup(admin_mc, remove_resource):
+    """ This ensures that roletemplates cleanup for clusters < v2.2.8
+        is performed correctly"""
+    client = admin_mc.client
+    rt = client.create_roleTemplate(name="rt-" + random_str())
+    remove_resource(rt)
+    assert rt.annotations["field.cattle.io/rtUpgrade"] == "true"
+
+    # create rt  without rancher api with a bad finalizer
+    api = CustomObjectsApi(admin_mc.k8s_client)
+    json = {
+        "apiVersion": "management.cattle.io/v3",
+        "kind": "RoleTemplate",
+        "metadata": {
+            "finalizers": [
+                "clusterscoped.controller.cattle.io/cluster-roletemplate-sync_fake"
+            ],
+            "name": "test-" + random_str(),
+        }
+    }
+    rt_k8s = api.create_cluster_custom_object(
+        group="management.cattle.io",
+        version="v3",
+        plural="roletemplates",
+        body=json,
+    )
+    remove_resource(rt_k8s["metadata"]["name"])
```

Comment:
Does this work ? I only see objects being passed, and it looks like this is a string.

Reply:
No, it won't work. Move it down to where you get the obj.